lib/Solver/Z3Solver.cpp

```diff
@@ -249,17 +249,24 @@ bool Z3SolverImpl::internalRunSolver(
 
   runStatusCode = SOLVER_RUN_STATUS_FAILURE;
 
-  for (ConstraintManager::const_iterator it = query.constraints.begin(),
-                                         ie = query.constraints.end();
-       it != ie; ++it) {
-    Z3_solver_assert(builder->ctx, theSolver, builder->construct(*it));
+  ConstantArrayFinder constant_arrays_in_query;
+  for(auto const& constraint : query.constraints)  {
+    Z3_solver_assert(builder->ctx, theSolver, builder->construct(constraint));
+    constant_arrays_in_query.visit(constraint);
   }
   ++stats::queries;
   if (objects)
     ++stats::queryCounterexamples;
 
   Z3ASTHandle z3QueryExpr =
       Z3ASTHandle(builder->construct(query.expr), builder->ctx);
+  constant_arrays_in_query.visit(query.expr);
+
+  for(auto const& constant_array : constant_arrays_in_query.results) {
+      for(auto const& arrayIndexValueExpr : builder->constant_array_assertions[constant_array]) {
+          Z3_solver_assert(builder->ctx, theSolver, arrayIndexValueExpr);
```

Comment:
You need to make a corresponding change to `Z3SolverImpl::getConstraintLog()`

Reply:
Ah missed that one, thanks! Will do.